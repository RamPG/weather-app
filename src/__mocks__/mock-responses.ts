import {
  GetGeoCityNameResponseType,
  GetGeoCoordsResponseType,
  GetWeatherResponseType,
} from '../types/response-types';

export const weatherResponseMock: GetWeatherResponseType = {
  lat: 25.5,
  lon: 24.4,
  timezone: '1200',
  timezone_offset: 3600,
  daily: [
    {
      dt: 5,
      sunrise: 5,
      sunset: 5,
      temp: {
        day: 25,
        min: 20,
        max: 27,
        night: 15,
        eve: 15,
        morn: 15,
      },
      feels_like: {
        day: 25,
        night: 21,
        eve: 23,
        morn: 24,
      },
      pressure: 10,
      humidity: 50,
      dew_point: 1,
      uvi: 1,
      clouds: 1,
      wind_speed: 34,
      wind_deg: 21,
      weather: [
        {
          id: 1,
          main: 'Clouds',
          icon: '04n',
        },
      ],
      pop: 25,
      rain: 15,
    },
  ],
  current: {
    dt: 5,
    sunrise: 5,
    sunset: 5,
    temp: 25,
    feels_like: 23,
    pressure: 10,
    humidity: 50,
    dew_point: 1,
    uvi: 1,
    clouds: 1,
    visibility: 1,
    wind_speed: 34,
    wind_deg: 21,
    weather: [
      {
        id: 1,
        main: 'Clouds',
        icon: '04n',
      },
    ],
  },
};

export const coordsResponseMock: GetGeoCoordsResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  class: '1',
  type: '1',
  importance: 1,
  icon: '1',
};

export const cityNameResponseMock: GetGeoCityNameResponseType = {
  place_id: '1',
  licence: '1',
  osm_type: '1',
  osm_id: '1',
  boundingbox: ['1'],
  lat: '25.5',
  lon: '24.4',
  display_name: 'Moscow, Russia',
  address: {
    city: 'Moscow',
    county: 'Moscow District',
    state: '',
    country: 'Russia',
    counry_code: '05',
  },
};
