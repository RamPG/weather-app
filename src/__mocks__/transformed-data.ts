import { DataCoordsStateType, TransformedCurrentDataType, TransformedDailyDataType } from '../types/state-types';

export const transformedDataCoords: DataCoordsStateType = {
  location: 'Moscow',
  latitude: 25.5,
  longitude: 24.4,
};

export const transformedDataCurrent: TransformedCurrentDataType = {
  imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
  temp: '-249째C',
  feelsLike: '-251째C',
  humidity: 50,
  weather: 'Clouds',
  windSpeed: 34,
};

export const transformedDataDaily: Array<TransformedDailyDataType> = [
  {
    id: 0,
    imgLink: 'https://openweathermap.org/img/wn/04n@2x.png',
    weekDayName: 'Friday',
    monthDay: 14,
    monthDayName: 'August',
    temp: {
      day: '-249째C',
      night: '-259째C',
    },
    weather: 'Clouds',
  },
];
