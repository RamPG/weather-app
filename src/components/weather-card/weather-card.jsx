import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import './weather-card.scss';

import { cityWeatherTodayFetch } from '../../actions/actions';
import withWeatherApi from '../../hoc-helpers';

const WeatherCard = ({
  cityWeatherTodayFetch, cityName, loading, error, data: {
    feelsLike, humidity, temp, weather, windSpeed, imgLink,
  },
}) => {
  useEffect(() => {
    cityWeatherTodayFetch();
  }, []);
  if (loading) {
    return <h1>Loading...</h1>;
  }
  if (error) {
    return <h1>Error!</h1>;
  }
  return (
    <section className="main-card">
      <img className="main-card__img" src={imgLink} />
      <table className="main-card__table">
        <tbody>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">
              Weather in
              {' '}
              {cityName}
            </th>
          </tr>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">Temperature</th>
            <td className="main-card__value">
              {temp}
              °C
            </td>
          </tr>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">Feels like</th>
            <td className="main-card__value">
              {feelsLike}
              °C
            </td>
          </tr>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">Humidity</th>
            <td className="main-card__value">
              {humidity}
              %
            </td>
          </tr>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">Weather</th>
            <td className="main-card__value">{weather}</td>
          </tr>
          <tr className="main-card__table-row">
            <th className="main-card__parameter">Wind speed</th>
            <td className="main-card__value">
              {windSpeed}
              {' '}
              m/s
            </td>
          </tr>
        </tbody>
      </table>

    </section>
  );
};
const mapStateToProps = ({ cityName, current: { data, loading, error } }) => ({
  cityName, data, loading, error,
});

const mapDispatchToProps = (dispatch, { weatherApi }) => ({
  cityWeatherTodayFetch: () => dispatch(cityWeatherTodayFetch(weatherApi)),
});
export default withWeatherApi(connect(mapStateToProps, mapDispatchToProps)(WeatherCard));
