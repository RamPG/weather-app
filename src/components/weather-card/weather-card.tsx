import React, { FunctionComponent } from 'react';

import './weather-card.scss';

import { TransformedCurrentDataType } from '../../types/state-types';

export const WeatherCard: FunctionComponent<{ data: TransformedCurrentDataType }> = ({
  data: {
    feelsLike, humidity, temp, weather, windSpeed, imgLink,
  },
}) => (
  <section>
    <p>
      {temp}
    </p>
    <p>
      <img src={imgLink} alt="weather" height="100" width="100" />
    </p>
    <p>
      {weather}
    </p>
    <p>
      Feels like
      {' '}
      {feelsLike}
    </p>
    <p>
      {windSpeed}
      {' '}
      m/s
    </p>
    <p>
      {humidity}
      %
    </p>
  </section>
);
