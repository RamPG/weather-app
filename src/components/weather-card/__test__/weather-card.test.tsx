import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { WeatherCard } from '../weather-card';
import { transformedDataCurrent } from '../../../__mocks__/transformed-data';

describe('Weather card', () => {
  afterEach(cleanup);
  it('Snapshot', () => {
    const { asFragment } = render(<WeatherCard data={transformedDataCurrent} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
