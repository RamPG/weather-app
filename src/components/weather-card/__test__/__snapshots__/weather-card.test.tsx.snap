// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather card Snapshot 1`] = `
<DocumentFragment>
  <section>
    <p>
      -249°C
    </p>
    <p>
      <img
        alt="weather"
        height="100"
        src="https://openweathermap.org/img/wn/04n@2x.png"
        width="100"
      />
    </p>
    <p>
      Clouds
    </p>
    <p>
      Feels like -251°C
    </p>
    <p>
      34 m/s
    </p>
    <p>
      50%
    </p>
  </section>
</DocumentFragment>
`;
